### DISCUSSION TITLE
IFUS Camera RTSP server freeze randomly

### DISCUSSION THREAD

**Discussion by ivanobressan** [roleoroleo/yi-hack-Allwinner-v2#561] [Q&A]
HI All,
i have 5 identical IFUS 108P outdoor cameras
3 of them works flawlessy, but 2 of them didn't works and i have problems wit rtsp server that stop streaming even if the process is still alive, if i kill the process and manually restrart it, i can see the stream for a few minutes, sometimes 5 sometimes 20 or 25 but after that, these 2 cams stop streaming even if i can see regulary the stream throug Yi home app.
The very strange things appens at night, if i reboot the cameras when outside is dark, i can see the stream during all the night with no issues, at the first morning lights the nightmare retart.
I've tried every kind of configuration, i replaced SD too, but nothing help.
Please help me

**Comment by Giuserver** [comment]
I had a similar problem. Solved like enable swap space (from web page). And using tcp on rtsp transport. like:
ffplay -rtsp_transport tcp -i "rtsp://ipcam/ch0_0.h264". Btw, it's seems a problem with connection no with camera. Try also to get closer to the camera.

**Reply by ivanobressan** [reply]
Hi, thansk for your replay.
TCP is enbled in frigate with this directive:
input_args: -rtsp_transport tcp
swap just enabled, hope this can solve, let's wait and see...

**Reply by roleoroleo** [reply]
Are the cam the same model?
I mean the suffix.

What happens if you move the running cam when the other cam doesn't work?
Maybe a power issue or a  network issue.

**Reply by ivanobressan** [reply]
Hi, i'm back after some days of tests, here some results and many other things to help paint the scene.
All the five cameras are same model with the same hack installed (h30ga)
all cameras are connected to the same 2.4Ghz outdoor access point and got 80 to 95 percent of signal
there is no way to get live555 rtsp server works, only the "alternative" works but with the same issues in 2 cameras out of 5
if i get in ssh to kill rtsp process and start again i get this:

root@CAMERA_SE:~# rtsp_server_yi -r low -p 554 -a alaw &
root@CAMERA_SE:~# Resolution: low
Audio: 2
Port: 554
User: 
Password: ********
Open /tmp/audio_fifo failed, disable audio.
Play URL: rtsp://127.0.0.1:554/ch0_1.h264
RTSP client connect, ip=192.168.0.10, port=36554 
RTSP client disconnect, ip=192.168.0.10, port=36554 
RTSP client connect, ip=192.168.0.10, port=58152 

time between connect and disconnect is totally random, maybe cameras works for minutes, hours or some days too, but freeze never appears when is dark outside (something related to IR sensor?).

A power issue can be the answer but consider that cameras newer reboots and are still reachable both via ssh and http even if the rtsp process crash.

Tanks again for your work.

**Reply by roleoroleo** [reply]
> Open /tmp/audio_fifo failed, disable audio.

This error is strange.
Maybe the hack is not properly installed.

**Reply by ivanobressan** [reply]
I replaced the sd cards thinking about a defect, so cards contains a fresh copy of your hack, how can i test if the camera firmware is corrupted? can i rollback from the start and redo every single step?

**Reply by roleoroleo** [reply]
Check the running processes with ps and post the output.

You should try to use the rtsp server based on live555.
Please run it in debug mode and post the output.

**Reply by fibilt** [reply]
how and where you run this: ffplay -rtsp_transport tcp -i "rtsp://ipcam/ch0_0.h264"? Thanks

**Comment by ivanobressan** [comment]
Nope, even swap didn't works...

**Reply by Giuserver** [reply]
Check if the camera restart when you lose connection. (you can check that from web page http://ipcam:8080/index.html in Uptime row)

**Reply by ivanobressan** [reply]
No, cameras never loose connection, ping always works and never report any error.

**Reply by Giuserver** [reply]
Can you try this just for a night:
Install FFmpeg on your computer 

First camera:
step 1) ffmpeg -hide_banner -rtsp_transport tcp -i "rtsp://**YOUR_IP_CAMERA1**/ch0_0.h264" -r 20 -vcodec copy -acodec copy -f segment -segment_time 600 -reset_timestamps 1 -segment_format mkv  -strftime 1 "camera1_video_%HH%MM%SS.mkv" &> log_camera1.txt";

Second camera:
step 1) shutdown IR, ONVIF, FTP, MQTT
step 2) ffmpeg -hide_banner -rtsp_transport tcp -i "rtsp://**YOUR_IP_CAMERA2**/ch0_0.h264" -r 20 -vcodec copy -acodec copy -f segment -segment_time 600 -reset_timestamps 1 -segment_format mkv  -strftime 1 "camera2_video_%HH%MM%SS.mkv" &> log_camera2.txt";

**Reply by ivanobressan** [reply]
HI, here i'm again after a long test session, i tried every single setup mix but got no result, talking about the test you asked me, i can say that the 2 logs doesn't contain any errors, here are one tail:

frame=968587 fps= 17 q=-1.0 size=N/A time=15:23:10.29 bitrate=N/A speed=   1x    
frame=968597 fps= 17 q=-1.0 size=N/A time=15:23:10.81 bitrate=N/A speed=   1x    
frame=968608 fps= 17 q=-1.0 size=N/A time=15:23:11.36 bitrate=N/A speed=   1x    
frame=968619 fps= 17 q=-1.0 size=N/A time=15:23:11.90 bitrate=N/A speed=   1x    
frame=968629 fps= 17 q=-1.0 size=N/A time=15:23:12.40 bitrate=N/A speed=   1x    
frame=968639 fps= 17 q=-1.0 size=N/A time=15:23:12.95 bitrate=N/A speed=   1x    
frame=968651 fps= 17 q=-1.0 size=N/A time=15:23:13.51 bitrate=N/A speed=   1x    
frame=968662 fps= 17 q=-1.0 size=N/A time=15:23:14.08 bitrate=N/A speed=   1x    
frame=968673 fps= 17 q=-1.0 size=N/A time=15:23:14.59 bitrate=N/A speed=   1x    

the last line you can see is the last line of the log.

i'm starting to think i bricked the cameras

**Reply by Giuserver** [reply]
sorry for a late response. I think you try to change sd (model and brand).. 
After a lost connection, you can reconnect normally? 
just try to ping the camera 24h and check if you have some package loss. (with no video capture active)

**Comment by grochib** [comment]
Hi, I have the exact same problem: one 1080 outdoor cam (IFUS) works flawlessly, the second 1080 outdoor cam (RFUS) is having the same problem: the stream interrupts for several seconds, the cam is still connected to wifi but reacts really slowly and then comes back as if nothing happened. According to the index page the cam stayed online the whole time. 
I changed the power supply and even tried the original one: it helped a bit but the problem kept on happening. Actually it seemed that the trick was turning it off and after several seconds back on again (not only rebooting!). 

I saw that your cam is streaming the low quality video AND audio (alaw). My first cam is only streaming low video without audio, the second one (the problematic cam) is streaming low video AND audio (alaw) - so just as your cam.

Now I'm wondering: is there maybe a connection between streaming audio and the random rtsp stream crashes? Maybe a memory overflow or something? 

On a sidenote: sometimes the stream freezes for a second and keeps on showing the same picture (the cam stays online though and does not appear to have been lost). But this only happens some movement is happening, like the mail man is coming. 

Of course I have swap enabled, snapshot disabled, movement detection disabled, cloud disabled, recording on sd-card disabled.  

Best

grochi

**Comment by chrbar11** [comment]
Hi @ivanobressan @Giuserver @grochib.

I have the same problem with my Yi Outdoor 1080p RFUS : randomly Yi-Hack Web server stops responding, the camera seems to lose the network connection!

Do you fix/mitigate these Yi Outdoor 1080p issues?
Do you please have any tips to help users of this camera model?

What Base Version (Yi firmware) and Yi-Hack Version are you running and recommending?

For my part, I did some tests but the camera continues to freeze.
I'm running Yi native firmware 9.0.26.07 with Yi-Hack version 0.2.6.
I tried with and without audio streaming enabled but I saw no improvement.
I tried with a dedicated SSID (2.4Ghz only) and an additional AP closer, but the problem persists.

I saw that this camera model seems to have a latency issue.
I did some tests with 3 cameras : 2 Yi Outdoor 1080p (camera1 and camera2) and 1 Yi 1080p Home (camera5).
All cameras are in the same room at the same distance from the router.
![YiOutdoor1080p-1](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/45576003/475db1b8-f3b0-44cf-8fe6-6afd35dcf9b3)

I'm newbie and I don't use a video surveillance software yet.
But I saw that this camera model has less latency issue when streaming.
I started live streaming with VLC on camera2 and switched streaming to camera3, and we can see the latency changing depending on the streaming activity.
![YiOutdoor1080p-2](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/45576003/8a77547b-54ab-47d1-8784-6066d7d2eda4)

I have seen people talking about the IFUS model.
What is the difference between the RFUS and IFUS model?

Thanks a lot for your help :)
Chris

**Reply by Giuserver** [reply]
seems a network problem.
first of all, test a camera for 24h under ping continuously and check eventually packet loss. After, the camera reboot it self?
also try to change sd. The correct hack it's in home page reported from roleoroleo (most of time if hack works for some hours, work every time, so it's a correct hack, for my experience). Try also with ffmpeg so you can logging. Last, after a lost connection, you can reconnect normally?

**Reply by chrbar11** [reply]
Thank you @Giuserver

This problem seems to relate to the network service of the camera only, the camera does not seem to stop or restart on its own (if I remember correctly from my past tests, the camera continues to record videos locally on the SD card).
I'm not sure, but I think the camera is coming back online on its own.

I'm going to do the 24h test to be sure.

**Reply by Giuserver** [reply]
Also try to capture for 10 minutes and stop for 1 minutes. And, again, capture 10 minutes and stop for 1 minute. Try all day like that and see if works great or if your problem persist.

**Comment by fibilt** [comment]
> I had a similar problem. Solved like enable swap space (from web page). And using tcp on rtsp transport. like:
> ffplay -rtsp_transport tcp -i "rtsp://ipcam/ch0_0.h264". Btw, it's seems a problem with connection no with camera. Try also to get closer to the camera.
